### ISSUE TITLE
Bun crashed when version specifier given for couple package

### ISSUE THREAD

**Issue by rootfebri** [oven-sh/bun#14518]
### How can we reproduce the crash?

REPRO:
```shell
bun add @tanstack/react-router:^1.57.13 @tanstack/router-devtools:^1.57.13
```

### Stack Trace (bun.report)

Bun v1.1.30 ([`7996d06`](<https://github.com/oven-sh/bun/tree/7996d06b8f5869ccc9ad9a421cd06469da4273fc>)) on windows x86_64 [AddCommand]

**panic**: Internal assertion failure

- [`output.zig:491`](<https://github.com/oven-sh/bun/blob/7996d06b8f5869ccc9ad9a421cd06469da4273fc/src/output.zig#L491>): `panic`
- [`bun.zig:3439`](<https://github.com/oven-sh/bun/blob/7996d06b8f5869ccc9ad9a421cd06469da4273fc/src/bun.zig#L3439>): `assertionFailure`
- [`install.zig:7968`](<https://github.com/oven-sh/bun/blob/7996d06b8f5869ccc9ad9a421cd06469da4273fc/src/install/install.zig#L7968>): `edit`
- [`install.zig:10429`](<https://github.com/oven-sh/bun/blob/7996d06b8f5869ccc9ad9a421cd06469da4273fc/src/install/install.zig#L10429>): `updatePackageJSONAndInstallWithManagerWithUpdates`
- [`install.zig:10126`](<https://github.com/oven-sh/bun/blob/7996d06b8f5869ccc9ad9a421cd06469da4273fc/src/install/install.zig#L10126>): `updatePackageJSONAndInstall`
- [`install.zig:8875`](<https://github.com/oven-sh/bun/blob/7996d06b8f5869ccc9ad9a421cd06469da4273fc/src/install/install.zig#L8875>): `updatePackageJSONAndInstallCatchError`
- [`cli.zig:1676`](<https://github.com/oven-sh/bun/blob/7996d06b8f5869ccc9ad9a421cd06469da4273fc/src/cli.zig#L1676>): `start`
- [`cli.zig:63`](<https://github.com/oven-sh/bun/blob/7996d06b8f5869ccc9ad9a421cd06469da4273fc/src/cli.zig#L63>): `start`
- [`main.zig:49`](<https://github.com/oven-sh/bun/blob/7996d06b8f5869ccc9ad9a421cd06469da4273fc/src/main.zig#L49>): `main`
- [`exe_common.inl:78`](<https://github.com/oven-sh/bun/blob/7996d06b8f5869ccc9ad9a421cd06469da4273fc/src/src/startup/exe_common.inl#L78>): `invoke_main`

**Comment by RiskyMH** [comment]
It looks like the correct syntax would be `bun add @tanstack/react-router@^1.57.13 @tanstack/router-devtools@^1.57.13`, which is also how it works on npm. We’ll take a closer look at why the syntax you tried is causing a crash, as Bun shouldn’t ever crash.

Also, @dylan-conway, should the `^` semver operator be reflected in the `package.json`? At the moment, it only records the version without the operator. And do you think we should support the syntax used in this case?

**Comment by rootfebri** [comment]
In my opinion it should reflect the package.json because when the `=` or `~` or `^` operator is given by the user it should install that version based on operator definition and the user is "know what you are doing". it might not used by most user but it should work as expected.
sorry for my bad english.

**Comment by dylan-conway** [comment]
> should the ^ semver operator be reflected in the package.json? At the moment, it only records the version without the operator. 

Yes, it should include `^` or `~` if they are included in the specifier unless `--exact` is used. It's a bug if it is not.

I believe the crash is a bug caused by adding two dependencies that fail to parse correctly. `:` is usually used for github dependencies (e.g. `github:dylan-conway/install-test`) so bun might be confusing these dependencies with github dependencies instead of erroring early
